src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectivenessRotors.hpp

```diff
@@ -79,8 +79,10 @@ class ActuatorEffectivenessRotors : public ModuleParams, public ActuatorEffectiv
 		bool three_dimensional_thrust_disabled{false}; ///< for handling of tiltrotor VTOL, as they pass in 1D thrust and collective tilt
 	};
 
+	/*** CUSTOM ***/
 	ActuatorEffectivenessRotors(ModuleParams *parent, AxisConfiguration axis_config = AxisConfiguration::Configurable,
-				    bool tilt_support = false);
+				    bool tilt_support = false, bool tilting_omnidir = false);
+	/*** CUSOTM ***/
```

Comment:
Have you seen that we already have a [`ActuatorEffectivenessTilts`](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectivenessTilts.hpp) that is used for [tiltrotors](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectivenessTiltrotorVTOL.hpp)? I think it would be more elegant if the tilting multirotor uses the Tilts, rather than hacking something into the Rotors

Reply:
Yes, I saw this file but here the problem is a bit different. In the file that you linked, the author just considered that the rotor can be tilted and with that the torque sign. Instead, in the case of tilting multicopters, the allocation matrix differs from the normal one and so I had to change a bit the code.
But  correct me if I'm wrong

Reply:
Looking at a closer look and talking to @sfuhrer I think you are right, but wouldn't it then be better to expand the tilt actuator to dynamically update the allocation matrix?

Reply:
Updating the allocation matrix with the tilt angle will lead to singularities in the allocation matrix. That's why the real allocation matrix is modified, resulting in a constant matrix with double the number of columns, which is always invertible (of course with the pseudo-inverse, being rectangular)